### PR TITLE
Add tickets endpoints

### PR DESCRIPTION
**Pull request by leosoubeste**

### CODE REVIEW

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  describe '.post_email_verification' do
+    let(:body_email) do
+      {
+        'result_url': 'http://myapp.com/callback',
+        'user_id': 'auth0|56604aa06bfc634d730d5959'
+      }
+    end
+    let(:email_verification) { client.post_email_verification(body_email) }
+    it { expect(email_verification).to include('ticket') }
```
Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb]: Can we add some unit tests for the case where the body is empty?

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let(:body_password) do
+      {
+        'result_url': 'http://myapp.com/callback',
+        'user_id': 'auth0|56604aa06bfc634d730d5959',
+        'new_password': 'secret'
+      }
+    end
+    let(:password_change) { client.post_password_change(body_password) }
+    it { expect(password_change).to include('ticket') }
+  end
```
Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb]: Can we add some unit tests for the case where the body is empty?

**Comment by ignaciojonas** [comment]
After rebase with master, and have Travis passing, we can merge this in our v2 branch.

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+describe Auth0::Api::V2::Tickets do
+  attr_reader :client, :user
+
+  before(:all) do
+    @client = Auth0Client.new(v2_creds)
+    username = Faker::Internet.user_name
+    email = "#{entity_suffix}#{Faker::Internet.safe_email(username)}"
+    password = Faker::Internet.password
+    connection = 'Username-Password-Authentication'
```
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb]: I see "Username-Password-Authentication" string spread all over the project. Should be create a global constant?

File: spec/lib/auth0/api/v2/tickets_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+describe Auth0::Api::V2::Tickets do
+  before :all do
+    dummy_instance = DummyClass.new
+    dummy_instance.extend(Auth0::Api::V2::Tickets)
+    @instance = dummy_instance
```
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/tickets_spec.rb]: Could lines 4,5,6 be merged in one line?

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      {
+        'result_url' => 'http://myapp.com/callback',
+        'user_id' => user['user_id'],
+        'new_password' => 'secret'
+      }
+    end
+    let(:password_change) { client.post_password_change(body_password) }
+    it { expect(password_change).to include('ticket') }
+  end
+end
```
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb]: Should we add a fail test?

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb (hunk 3 above)
Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb]: There is one already in place in authentication_endpoints and  can be used: `connection = Auth0::Api::AuthenticationEndpoints::UP_AUTH,`. What do you think?

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb (hunk 5 above)
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb:43]: Is it posible to get an error in the response? Should we treat/ test it?

Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb:43]: We are testing the possible error in the unit test.- We'll keep the integration tests just for the happy flow.

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb (hunk 3 above)
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb]: Agree

File: spec/integration/lib/auth0/api/v2/api_tickets_spec.rb (hunk 5 above)
Comment (falvarisqueta) [inline spec/integration/lib/auth0/api/v2/api_tickets_spec.rb:43]: Ok

**Comment by falvarisqueta** [comment]
Looks good to me
